### ISSUE TITLE
Question: How can I revert parameter indexes to work like 0.8.x

### ISSUE THREAD

**Issue by ponny** [lostisland/faraday#474]
Here's some example code:

``` rb
uri = 'http://example.com?bar%5B0%5D=foo'
connection = Faraday.default_connection
request = connection.build_request('delete') do |req|
            req.url(Addressable::URI.parse(uri).normalize.to_s)
            req.headers = Faraday::Utils::Headers.new({})
            req.body = nil
          end

request.to_env(connection).url
```

In < 0.8, the result is "http://example.com/?bar%5B0%5D=foo"
In >= 0.9 the result is "http://example.com/?bar%5B%5D=foo"

The change has broken one of our libraries.  Is there any way to configure Faraday to act the old way?

**Comment by mislav** [comment]
Sorry that you had trouble. Yes, the current situation is that you will lose the index number. In future minor version I plan to drop support for index numbers altogether. In your case that will help, because your original parameter will simply be left unchanged instead of losing the `0`.

**Comment by iMacTia** [comment]
Closing as we're not going to support indexes.
Only solution here is to fix faraday to v0.8 on the Gemfile
